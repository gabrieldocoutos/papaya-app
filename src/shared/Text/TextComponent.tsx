import React from 'react';
import {Text as RNText} from 'react-native';

console.log('test commit');

const Text = ({children}) => {
  return <RNText>{children}</RNText>;
};

export {Text};
