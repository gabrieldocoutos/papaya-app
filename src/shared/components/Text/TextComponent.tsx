import React from "react";
import styled from "styled-components/native";
import {
  color,
  ColorProps,
  space,
  SpaceProps,
  fontSize,
  FontSizeProps,
  fontFamily,
  FontFamilyProps,
  typography,
  TypographyProps,
  layout,
  LayoutProps,
} from "styled-system";

const StyledText = styled.Text<
  ColorProps & SpaceProps & FontSizeProps & FontFamilyProps & TypographyProps & LayoutProps
>`
  ${color};
  ${space};
  ${fontSize}
  ${fontFamily}
  ${typography}
  ${layout}
`;

const buildFontWeight = (fontWeight: TextProps["fontFamily"]) => {
  switch (fontWeight) {
    case "Thin":
    case "Thin_Italic":
      return 100;
    case "Light":
    case "Light_Italic":
      return 300;
    case "Regular":
    case "Regular_Italic":
      return 400;
    case "Bold":
    case "Bold_Italic":
      return 700;
    case "Black":
    case "Black_Italic":
      return 900;
  }
};

const Text = ({
  children,
  color = "primary",
  fontFamily = "Regular",
  fontSize = 16,
  ...rest
}: TextProps) => {
  const font = `Lato_${buildFontWeight(fontFamily) + fontFamily}`;
  return (
    <StyledText fontFamily={font} color={color} fontSize={fontSize} {...rest}>
      {children}
    </StyledText>
  );
};

export interface TextProps
  extends ColorProps,
    SpaceProps,
    FontSizeProps,
    TypographyProps,
    LayoutProps {
  children: string;
  fontFamily?:
    | "Thin"
    | "Thin_Italic"
    | "Light"
    | "Light_Italic"
    | "Regular"
    | "Regular_Italic"
    | "Bold"
    | "Bold_Italic"
    | "Black"
    | "Black_Italic";
}

export { Text };
