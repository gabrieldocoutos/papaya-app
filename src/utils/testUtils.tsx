import React, { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react-native";
import { ThemeProvider } from "styled-components/native";
import { theme } from "shared/configuration";

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react-native";
export { customRender as render };
