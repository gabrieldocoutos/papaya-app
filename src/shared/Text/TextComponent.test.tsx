import React from "react";
import { render } from "utils/testUtils";

import { Text } from "shared/Text/TextComponent";

test("snapshot with regular fontWeight", () => {
  const { toJSON } = render(<Text color="primary">Lato Black</Text>);
  expect(toJSON()).toMatchSnapshot();
});

test("snapshot with Thin fontWeight", () => {
  const { toJSON } = render(
    <Text color="primary" fontFamily="Thin">
      Lato Black
    </Text>
  );
  expect(toJSON()).toMatchSnapshot();
});

test("snapshot with Light fontWeight", () => {
  const { toJSON } = render(
    <Text color="primary" fontFamily="Light">
      Lato Black
    </Text>
  );
  expect(toJSON()).toMatchSnapshot();
});

test("snapshot with Bold fontWeight", () => {
  const { toJSON } = render(
    <Text color="primary" fontFamily="Bold">
      Lato Black
    </Text>
  );
  expect(toJSON()).toMatchSnapshot();
});

test("snapshot with Black fontWeight", () => {
  const { toJSON } = render(
    <Text color="primary" fontFamily="Black">
      Lato Black
    </Text>
  );
  expect(toJSON()).toMatchSnapshot();
});
