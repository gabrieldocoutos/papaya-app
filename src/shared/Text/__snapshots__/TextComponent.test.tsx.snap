// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot with Black fontWeight 1`] = `
<Text
  color="primary"
  fontFamily="Lato_900Black"
  fontSize={16}
  style={
    Array [
      Object {
        "color": "green",
        "fontFamily": "Lato_900Black",
        "fontSize": 16,
      },
    ]
  }
>
  Lato Black
</Text>
`;

exports[`snapshot with Bold fontWeight 1`] = `
<Text
  color="primary"
  fontFamily="Lato_700Bold"
  fontSize={16}
  style={
    Array [
      Object {
        "color": "green",
        "fontFamily": "Lato_700Bold",
        "fontSize": 16,
      },
    ]
  }
>
  Lato Black
</Text>
`;

exports[`snapshot with Light fontWeight 1`] = `
<Text
  color="primary"
  fontFamily="Lato_300Light"
  fontSize={16}
  style={
    Array [
      Object {
        "color": "green",
        "fontFamily": "Lato_300Light",
        "fontSize": 16,
      },
    ]
  }
>
  Lato Black
</Text>
`;

exports[`snapshot with Thin fontWeight 1`] = `
<Text
  color="primary"
  fontFamily="Lato_100Thin"
  fontSize={16}
  style={
    Array [
      Object {
        "color": "green",
        "fontFamily": "Lato_100Thin",
        "fontSize": 16,
      },
    ]
  }
>
  Lato Black
</Text>
`;

exports[`snapshot with regular fontWeight 1`] = `
<Text
  color="primary"
  fontFamily="Lato_400Regular"
  fontSize={16}
  style={
    Array [
      Object {
        "color": "green",
        "fontFamily": "Lato_400Regular",
        "fontSize": 16,
      },
    ]
  }
>
  Lato Black
</Text>
`;
